fhirclient/models/fhirsearch.py

```diff
@@ -124,14 +128,35 @@ def perform(self, server):
         bundle = bundle.Bundle(res)
         bundle.origin_server = server
         return bundle
-    
+
+    # Use forward references to avoid circular imports
+    def perform_iter(self, server) -> Iterator['Resource']:
+        """ Perform the search by calling `perform` and return an iterator that yields
+        Bundle instances.
+
+        :param server: The server against which to perform the search
+        :returns: An iterator of Bundle instances
+        """
+
+        first_bundle = self.perform(server)
+        if not first_bundle:
+            return iter([])
+        yield first_bundle
+
     def perform_resources(self, server):
         """ Performs the search by calling `perform`, then extracts all Bundle
         entries and returns a list of Resource instances.
         
         :param server: The server against which to perform the search
         :returns: A list of Resource instances
         """
+        # Old method with deprecation warning
+        warnings.warn(
+            "perform_resources() is deprecated and will be removed in a future release. "
+            "Please use perform_resources_iter() instead.",
+            DeprecationWarning,
+        )
+
         bundle = self.perform(server)
         resources = []
         if bundle is not None and bundle.entry is not None:
```

Comment:
nit: this code could just be `return list(self.perform_resources_iter())` yeah? (just to reduce duplicated code paths, and to emphasize that the method is now just a "worse" version of the _iter one)

Reply:
Since you'd prefer to change a behavior of perfom_resources(), my current implementations of perform_iter() and perform_resources_iter() depend on perform(). I can refactor perform_iter() to remove this dependency  while  keeping both perform() and perfom_resources() with DeprecationWarning to maintain backward compatibility.  Please, let me know what direction you'd prefer here.

Reply:
Yeah it sounds reasonable to have the deprecation warning in both perform & perform_resources. Refactoring could maybe be as simple as moving the guts out to a helper method like `_perform_perform` that both call (don't name it that :smile:)